staging/src/k8s.io/client-go/transport/token_source.go

```diff
@@ -72,14 +93,20 @@ func NewCachedTokenSource(ts oauth2.TokenSource) oauth2.TokenSource {
 type tokenSourceTransport struct {
 	base http.RoundTripper
 	ort  http.RoundTripper
+	src  ResettableTokenSource
 }
 
 func (tst *tokenSourceTransport) RoundTrip(req *http.Request) (*http.Response, error) {
 	// This is to allow --token to override other bearer token providers.
 	if req.Header.Get("Authorization") != "" {
 		return tst.base.RoundTrip(req)
 	}
-	return tst.ort.RoundTrip(req)
+	resp, err := tst.ort.RoundTrip(req)
```

Comment:
at the round-tripper level, wouldn't a 401 error come back as a resp (with code 401) and a nil `err`? I thought `err` was reserved for conditions that prevented us from making the request

Reply:
that's right